src/Columns/ColumnNullable.h

```diff
@@ -62,14 +62,19 @@ class ColumnNullable final : public COWHelper<IColumn, ColumnNullable>
 
     /**
      * If isNullAt(n) returns false, returns the nested column's getDataAt(n), otherwise returns a special value
-     * EMPTY_STRING_REF indicating that data is not present.
+     * with size == 0 indicating that data is not present.
      */
     StringRef getDataAt(size_t n) const override
     {
+        auto nestedData = getNestedColumn().getDataAt(n);
         if (isNullAt(n))
-            return EMPTY_STRING_REF;
+        {
```

Comment:
I think this branch can be removed.

Reply:
In that case we'll get non-zero size `StringRef`. E.g. 4 for UInt32, and IMO that would lead to matching hashes for UInt32(0) and null item, which might break something else.